packages/extension-font-family/src/font-family.ts

```diff
@@ -72,9 +72,15 @@ export const FontFamily = Extension.create<FontFamilyOptions>({
           .setMark('textStyle', { fontFamily })
           .run()
       },
-      unsetFontFamily: () => ({ chain }) => {
+      unsetFontFamily: () => ({ chain, tr }) => {
+        const { selection } = tr
+
+        // Retrieve all previous mark attributes applied to the current selection
+        const previousMarkAttributes = selection.$anchor.marks().map(mark => mark.attrs)
```

Comment:
Wait, is this right? Don't we only want the `textStyle` marks, not necessarily all mark attributes?

Reply:
So, fontFamily, isn't really a mark itself, it is just an additional attribute on the textStyle mark. 

What I'm saying is that if there were a mark that had attributes (that was not textStyle), for example the highlight extension can store it's color also in attribute. This would be trying to merge that attribute into the textStyle extension which is incorrect

Reply:
Oooooh, got it :)